compiler/rustc_mir/src/dataflow/framework/engine.rs

```diff
@@ -208,12 +208,11 @@ where
             }
         }
 
-        let mut state = analysis.bottom_value(body);
         while let Some(bb) = dirty_queue.pop() {
             let bb_data = &body[bb];
 
             // Apply the block transfer function, using the cached one if it exists.
-            state.clone_from(&entry_sets[bb]);
+            let mut state = entry_sets[bb].clone();
```

Comment:
IIUC `state.clone_from(...)` and `state = (...).clone()` should be equivalent in functionality, so this isn't really a change..

Reply:
if we have a domain which potentially reuses an allocation when using `clone_from` this could matter, but I don't really know if that can be the case

Reply:
Doesn't `BitSet` already do that?

Reply:
Should we just do a perf run to check?

Reply:
I'd rather not remove this optimization, even if it doesn't affect perf *today*

Reply:
Hmm. I found this code confusing when I read through it since it makes it look like `state` is shared between each loop iteration, when in fact it's reinitialized for each iteration.